packages/admin/src/components/ReferenceArrayField.js

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { ReferenceArrayField as RAReferenceArrayField } from 'react-admin';
+
+const ReferenceArrayField = ({ record, basePath, children, label, source, reference }) => {
+	if (record && record[source]) {
+		record[source] = record[source].map((i) => i.id || i);
```

Comment:
Podemos tentar trabalhar com a imutabilidade aqui.

```suggestion
		const newRecord = { ...record, [source]: record[source].map((i) => i.id || i) }
```

Reply:
Não consegui fazer essa sugestão funcionar

Reply:
Eu acabei não fazendo isso no PR, mas, vc chegou passar esse `newRecord` pro componente no lugar do `record`? Eu fiz aqui e foi de boa

Reply:
Funcionou!